scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -513,6 +521,199 @@ private void satisfyGoalGeneral(Goal goal) throws SchedulingInterruptedException
     logger.info("Finishing goal satisfaction for goal " + goal.getName() +":"+ (missingConflicts.size() == 0 ? "SUCCESS" : "FAILURE. Number of conflicts that could not be addressed: " + missingConflicts.size()));
   }
 
+  private ConflictSolverResult solveMissingRecurrenceConflict(
+      final MissingRecurrenceConflict missingRecurrenceConflict,
+      final Goal goal
+      ) throws SchedulingInterruptedException
+  {
+    Optional<Long> maxIterations = Optional.empty();
+    final var spaceToFill = (missingRecurrenceConflict.nextStart.minus(missingRecurrenceConflict.lastStart));
+    final var remainderMin = spaceToFill.remainderOf(missingRecurrenceConflict.minMaxConstraints.end);
+    final var remainderMax = spaceToFill.remainderOf(Duration.max(missingRecurrenceConflict.minMaxConstraints.start, MICROSECOND));
+    final var minNbActivities = spaceToFill.dividedBy(missingRecurrenceConflict.minMaxConstraints.end) - 1 + (remainderMin.isZero() ? 0 : 1);
+    final var maxNbActivities = spaceToFill.dividedBy(Duration.max(missingRecurrenceConflict.minMaxConstraints.start, MICROSECOND)) - 1+ (remainderMax.isZero() ? 0 : 1);
+    final var step = maxIterations.isPresent() ? (int) Math.floor((maxNbActivities - minNbActivities)/maxIterations.get()) : 1;
+    var nbActivities = minNbActivities;
+    List<SchedulingActivity> lastPartiallySat = new ArrayList<>();
+    var satisfaction = ConflictSatisfaction.NOT_SAT;
+    //conflict production should be easy as it involves static constraints only.
+    //we limit how many times it can fail before bailing
+    var conflictProductionHasNotProgressedSince = 0;
+    var maxNoProgress = 10;
+    while(true){
+      //create network with nbActivities
+      final var conflict = makeActivityNetworkConflict(missingRecurrenceConflict, nbActivities, lastPartiallySat);
+      if(conflict.isPresent()) {
+        conflictProductionHasNotProgressedSince = 0;
+        final var conflictResult = solveActivityNetworkConflict(conflict.get(), goal, ScheduleAt.LATEST);
+        switch (conflictResult.satisfaction()) {
+          case ConflictSatisfaction.SAT:
+            return conflictResult;
+          case ConflictSatisfaction.PARTIALLY_SAT:
+            lastPartiallySat = conflictResult.activitiesCreated().stream().sorted(Comparator.comparing(
+                SchedulingActivity::startOffset)).toList();
+            satisfaction = ConflictSatisfaction.PARTIALLY_SAT;
+            break;
+          default:
+        }
+        if(hasDuplicates(conflictResult.activitiesCreated().stream().toList())){
+          //duplicates means it is unsolvable and it is no use increasing the number of activities
+          break;
+        }
+      } else {
+        conflictProductionHasNotProgressedSince++;
+      }
+      if(nbActivities < maxNbActivities && conflictProductionHasNotProgressedSince < maxNoProgress){
```

Comment:
Does `makeActivityNetworkConflict` use a simulation, or do some other expensive operation? If not, I don't see the purpose of using `conflictProductionHasNotProgressedSince` to limit the number of times it can try to create the network.

Reply:
no expensive operation but it could loop a lot (when the user is setting the min separation to 0, the number of iterations is going to be `separatedByAtMost.in(MICROSECONDS)`) if `makeActivityNetworkConflict` is failing repeateadly. At some point, we know it won't progress anymore.